modules/light-clients/07-tendermint/light_client_module.go

```diff
@@ -21,19 +22,13 @@ type LightClientModule struct {
 }
 
 // NewLightClientModule creates and returns a new 07-tendermint LightClientModule.
-func NewLightClientModule(cdc codec.BinaryCodec, authority string) LightClientModule {
+func NewLightClientModule(key storetypes.StoreKey, cdc codec.BinaryCodec, authority string) LightClientModule {
 	return LightClientModule{
-		keeper: keeper.NewKeeper(cdc, authority),
+		keeper:        keeper.NewKeeper(cdc, authority),
+		storeProvider: clienttypes.NewStoreProvider(key),
 	}
```

Comment:
what do you think about passing `storeProvider` as an argument instead of the key? create once in app.go and can pass to many light client modules.

Also a consideration could be to return it from the `IBCKeeper` from some accessor method..

Reply:
Updated in https://github.com/cosmos/ibc-go/pull/6028/commits/769f51467909ffa5fcac28cc6cf5b0c74f5c3db3.